sklearn/metrics/tests/test_regression.py

```diff
@@ -330,3 +352,90 @@ def test_mean_absolute_percentage_error():
     y_true = random_number_generator.exponential(size=100)
     y_pred = 1.2 * y_true
     assert mean_absolute_percentage_error(y_true, y_pred) == pytest.approx(0.2)
+
+
+@pytest.mark.parametrize("distribution",
+                         ["normal", "lognormal", "exponential", "uniform"])
+@pytest.mark.parametrize("target_quantile", [0.05, 0.5, 0.75])
+def test_mean_pinball_loss_on_constant_predictions(
+    distribution,
+    target_quantile
+):
+    if not hasattr(np, "quantile"):
+        pytest.skip("This test requires a more recent version of numpy "
+                    "with support for np.quantile.")
+
+    # Check that the pinball loss is minimized by the empirical quantile.
+    n_samples = 1000
+    rng = np.random.RandomState(42)
+    data = getattr(rng, distribution)(size=n_samples)
+
+    # Compute the best possible pinball loss for any constant predictor:
+    best_pred = np.quantile(data, target_quantile)
+    best_constant_pred = np.full(n_samples, fill_value=best_pred)
+    best_pbl = mean_pinball_loss(data, best_constant_pred,
+                                 alpha=target_quantile)
+
+    # Evaluate the loss on a grid of quantiles
+    candidate_predictions = np.quantile(data, np.linspace(0, 1, 100))
+    for pred in candidate_predictions:
+        # Compute the pinball loss of a constant predictor:
+        constant_pred = np.full(n_samples, fill_value=pred)
+        pbl = mean_pinball_loss(data, constant_pred, alpha=target_quantile)
+
+        # Check that the loss of this constant predictor is greater or equal
+        # than the loss of using the optimal quantile (up to machine
+        # precision):
+        assert pbl >= best_pbl - np.finfo(best_pbl.dtype).eps
+
+        # Check that the value of the pinball loss matches the analytical
+        # formula.
+        expected_pbl = (
+            (pred - data[data < pred]).sum() * (1 - target_quantile) +
+            (data[data >= pred] - pred).sum() * target_quantile
+        )
+        expected_pbl /= n_samples
+        assert_almost_equal(expected_pbl, pbl)
+
+    # Check that we can actually recover the target_quantile by minimizing the
+    # pinball loss w.r.t. the constant prediction quantile.
+    def objective_func(x):
+        if x < 0 or x > 1:
+            return np.inf
+        pred = np.quantile(data, x)
+        constant_pred = np.full(n_samples, fill_value=pred)
+        return mean_pinball_loss(data, constant_pred, alpha=target_quantile)
```

Comment:
```suggestion
        return mean_pinball_loss(data, x, alpha=target_quantile)
```

Reply:
This cannot work, x cannot be a scalar if data is any array: mean_pinball_loss will check that the two argument have matching shapes. I will fix in my next push.

Reply:
Done in 8edbed1 . Is this what you intended?

Reply:
Exactly.